pkg/services/app/commit.go

```diff
@@ -0,0 +1,75 @@
+package app
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/fluxcd/go-git-providers/github"
+	"github.com/fluxcd/go-git-providers/gitprovider"
+	"github.com/weaveworks/weave-gitops/pkg/gitproviders"
+	"github.com/weaveworks/weave-gitops/pkg/utils"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+type CommitParams struct {
+	Name             string
+	Namespace        string
+	GitProviderToken string
+}
+
+func (a *App) GetCommits(params CommitParams) ([]gitprovider.Commit, error) {
+	ctx := context.Background()
+
+	app, err := a.kube.GetApplication(ctx, types.NamespacedName{Name: params.Name, Namespace: params.Namespace})
+	if err != nil {
+		return nil, err
+	}
+
+	owner, err := utils.GetOwnerFromUrl(app.Spec.URL)
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve owner: %w", err)
+	}
+
+	gitProvider, err := a.gitProviderFactory(params.GitProviderToken)
+	if err != nil {
+		return nil, err
+	}
+
+	accountType, err := gitProvider.GetAccountType(owner)
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve account type: %w", err)
+	}
+
+	var commits []gitprovider.Commit
+
+	if accountType == gitproviders.AccountTypeUser {
+		userRepoRef := gitproviders.NewUserRepositoryRef(github.DefaultDomain, owner, params.Name)
+		commits, err = gitProvider.GetCommitsFromUserRepo(userRepoRef, app.Spec.Branch)
+		if err != nil {
+			return nil, fmt.Errorf("unable to get Commits for user repo: %s", err)
+		}
+	} else {
+		repoName := utils.UrlToRepoName(app.Spec.URL)
+		orgRepoRef := gitproviders.NewOrgRepositoryRef(github.DefaultDomain, owner, repoName)
```

Comment:
it seems like we could use the same variable to pass in the repository name to `NewUserRepositoryRef` and `NewOrgRepositoryRef` in lines 47 and 54